soh/include/macros.h

```diff
@@ -2,6 +2,7 @@
 #define MACROS_H
 
 #include <endianness.h>
+#include "z64.h"
```

Comment:
In current this file does something naughty and refers to GraphicsContext without properly including it's file, it probably just worked by chance. Unfortunately me including Macros.h in CustomMessageTypes (to properly set TEXT_SHOP_ITEM_RANDOM_CONFIRM) must have changed compilation order and made this fail to build, so I had to actually include the file it is from to build otherwise I get a `unknown type name ‘GraphicsContext’` exception.

Reply:
In the file where you use `macros.h` is where you would want to include the header that has graphics context before it. You should not be adding to this header file, it is purposefully meant to depend on other headers externally.

Reply:
Decomp does currently have the struct forward declared at least, so yea we could do that too.

Reply:
Adding  '#include "z64.h"' to CustomMessageType did NOT fix the issue, which is expected as the newly used macros don't use the graphics. 

I am pushing a known broken state to see what the tests do.